pkg/machine/wsl/util_windows.go

```diff
@@ -311,30 +245,25 @@ func addRunOnceRegistryEntry(command string) error {
 
 func encodeUTF16Bytes(s string) []byte {
 	u16 := utf16.Encode([]rune(s))
-	u16le := make([]byte, len(u16)*2)
-	for i := 0; i < len(u16); i++ {
-		u16le[i<<1] = byte(u16[i])
-		u16le[(i<<1)+1] = byte(u16[i] >> 8)
+	buf := new(bytes.Buffer)
+	for _, r := range u16 {
+		_ = binary.Write(buf, binary.LittleEndian, r)
 	}
-	return u16le
+	return buf.Bytes()
 }
```

Comment:
The functions of both are exactly the same. I wrote a comparison code to demonstrate this: https://go.dev/play/p/lcmCqSuMRnh